ReferencePluginC/ControlC.cs

```diff
@@ -0,0 +1,85 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Data;
+using System.Drawing;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using System.Windows.Forms;
+
+using Paratext.PluginInterfaces;
+
+
+namespace ReferencePluginC
+{
+	public partial class ControlC : EmbeddedPluginControl
+	{
+		private IVerseRef m_Reference;
+		private IProject m_Project;
+		IWindowPluginHost m_Host;
+
+		public ControlC()
+		{
+			InitializeComponent();
+		}
+		public override void OnAddedToParent(IPluginChildWindow parent, IWindowPluginHost host, string state)
+		{
+			parent.SetTitle(PluginC.pluginName);
+			m_Host = host;
+			m_Project = parent.CurrentState.Project;
+			m_Reference = parent.CurrentState.VerseRef;
+
+			parent.ProjectChanged += ProjectChanged;
+			parent.VerseRefChanged += VerseRefChanged;
+
+			ShowCurrentInfo();
+		}
+
+		private void ShowCurrentInfo()
+		{
+			List<string> lines = new List<string>();
+			lines.Add($"Project name: {m_Project.ShortName}");
+			lines.Add($"Current Book: {m_Reference.BookCode}");
+			lines.Add($"Current Chapter: {m_Reference.ChapterNum}");
+			lines.Add($"Current Verse: {m_Reference.VerseNum}");
+			textBox.Lines = lines.ToArray();
+		}
+
+		public override string GetState()
+		{
+			return null;
+		}
+
+		public override void DoLoad(IProgressInfo progressInfo)
+		{
+		}
+
+		public void ProjectChanged(IPluginChildWindow sender, IProject newProject)
+		{
+			m_Project = newProject;
+			m_Reference = sender.CurrentState.VerseRef;
+			ShowCurrentInfo();
+		}
+
+		public void VerseRefChanged(IPluginChildWindow sender, IVerseRef oldReference, IVerseRef newReference)
+		{
+			m_Reference = newReference;
+			ShowCurrentInfo();
+		}
+
+		public void NextVerse(Object sender, EventArgs e)
+		{
+			m_Reference = m_Reference.GetNextVerse(m_Project);
+			var syncGroup = m_Host.ActiveWindowState.SyncReferenceGroup;
```

Comment:
This is probably technically correct, since the plugin window will probably always be the active window. However, there is possibly the ever-so-slight chance that some other window could become active between the time the menu item is clicked and the time this event is processed. Even though that's probably impossible -- I think this will always happen on the UI thread -- it would probably make more logical sense to hold onto this control's parent (IPluginChildWindow) and then access its CurrentState to get the SyncReferenceGroup. That way it's guaranteed to be correct and would probably be more helpful at illustrating the correct usage for some future plugin developer who may be doing something that is not necessarily on the UI thread.